### PR TITLE
FR-1868 Contested RespSol update details changes

### PR DESCRIPTION
**Pull request by ravinderpal**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FR-1868

### Change description ###

Respondent Solicitor can edit Respondent details and Respondent Solicitor details. 

Functionality put behind a feature toggle.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by johnoneill46 — APPROVED** [review]

File: definitions/contested/json/CaseEventToFields/CaseEventToFields.json [hunk 1]
```diff
@@ -7259,5 +7259,209 @@
...
+    "CaseTypeID": "FinancialRemedyContested",
+    "CaseEventID": "FR_updateRespondentContactDetails",
+    "CaseFieldID": "respondentRepresented",
+    "PageFieldDisplayOrder": 2,
+    "DisplayContext": "MANDATORY",
+    "PageID": 2,
+    "PageDisplayOrder": 2,
+    "PageColumnNumber": 1,
+    "ShowSummaryChangeOption": "Y"
+  },
```
Comment (ocarleton) [inline definitions/contested/json/CaseEventToFields/CaseEventToFields.json:7310]: I think we shouldn't include this in the  FR_updateRespondentContactDetails event

File: definitions/contested/json/CaseEventToFields/CaseEventToFields.json [hunk 2]
```diff
@@ -7259,5 +7259,209 @@
...
+    "PageDisplayOrder": 2,
+    "PageColumnNumber": 1,
+    "FieldShowCondition": "respondentRepresented=\"Yes\"",
+    "ShowSummaryChangeOption": "Y"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "FinancialRemedyContested",
+    "CaseEventID": "FR_updateRespondentContactDetails",
+    "CaseFieldID": "rRespondentLabel2",
```
Comment (ocarleton) [inline definitions/contested/json/CaseEventToFields/CaseEventToFields.json:7419]: We need to add auth for this field rRespondentLabel2
RU should be enough

File: definitions/contested/json/CaseEventToFields/CaseEventToFields.json [hunk 3]
```diff
@@ -7259,5 +7259,209 @@
     "PageDisplayOrder": 1,
     "PageColumnNumber": 1,
     "ShowSummaryChangeOption": "Y"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "FinancialRemedyContested",
+    "CaseEventID": "FR_updateRespondentContactDetails",
+    "CaseFieldID": "respondentDetailsLabel",
```
Comment (ocarleton) [inline definitions/contested/json/CaseEventToFields/CaseEventToFields.json:7267]: i think we need to add a R permission for respondentDetailsLabel

File: definitions/contested/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json [hunk 4]
```diff
@@ -1272,5 +1272,12 @@
...
     "UserRole": "caseworker-divorce-financialremedy-judiciary",
     "CRUD": "R"
+  },
+  {
+    "LiveFrom": "01/09/2020",
+    "CaseTypeID": "FinancialRemedyContested",
+    "CaseEventID": "FR_updateRespondentContactDetails",
+    "UserRole": "[RESPSOLICITOR]",
+    "CRUD": "CRU"
   }
```
Comment (ocarleton) [inline definitions/contested/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json:1282]: We have a failing test as events need R permission atleast for each of the other roles as well
(solicitor, judge, caseworker). So can we add in:

{
    "LiveFrom": "01/09/2020",
    "CaseTypeID": "FinancialRemedyContested",
    "CaseEventID": "FR_updateRespondentContactDetails",
    "UserRole": "caseworker-divorce-financialremedy-courtadmin",
    "CRUD": "R"
  },
  {
    "LiveFrom": "01/09/2020",
    "CaseTypeID": "FinancialRemedyContested",
    "CaseEventID": "FR_updateRespondentContactDetails",
    "UserRole": "caseworker-divorce-financialremedy-solicitor",
    "CRUD": "R"
  },
  {
    "LiveFrom": "01/09/2020",
    "CaseTypeID": "FinancialRemedyContested",
    "CaseEventID": "FR_updateRespondentContactDetails",
    "UserRole": "caseworker-divorce-financialremedy-judiciary",
    "CRUD": "R"
  },

File: definitions/contested/json/AuthorisationCaseEvent/AuthorisationCaseEvent-respSol-nonprod.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    "CaseEventID": "FR_updateRespondentContactDetails",
+    "UserRole": "[RESPSOLICITOR]",
+    "CRUD": "CRU"
+  },
+  {
+  "LiveFrom": "01/09/2020",
+  "CaseTypeID": "FinancialRemedyContested",
+  "CaseEventID": "FR_updateRespondentContactDetails",
+  "UserRole": "caseworker-divorce-financialremedy-judiciary",
+  "CRUD": "CRU"
```
Comment (ocarleton) [inline definitions/contested/json/AuthorisationCaseEvent/AuthorisationCaseEvent-respSol-nonprod.json:14]: These should be 
"CRUD": "R"
for all except [RESPSOLICITOR] who is CRU. As only RESPSOLICITOR should be able to trigger this event, the others only need R for seeing it in history

File: package.json [hunk 6]
```diff
@@ -99,7 +99,6 @@
     "bl": "4.0.3",
     "eslint-utils": "^1.4.1",
     "https-proxy-agent": "2.2.3",
-    "lodash": "4.17.19",
-    "bl": "4.0.3"
+    "lodash": "4.17.19"
```
Comment (shaunganley) [inline package.json:102]: any reason for this?

**Review by shaunganley — APPROVED** [review]

File: package.json (hunk 6 above)
Comment (ravinderpal) [reply package.json:102]: duplicate of line 99

Comment (ravinderpal) [reply package.json:102]: duplicate of line 99

**Pull request merged**